### PR TITLE
Add adal as dependency to core

### PR DESCRIPTION
**Pull request by williexu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/setup.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 
 # TODO These dependencies should be updated to reflect only what this package needs
 DEPENDENCIES = [
+    'adal>=1.0.0',
```
Comment (yugangw-msft) [inline src/azure-cli-core/setup.py]: Have we figured out why we end up with lower version?
If this issue is only repro'd under dev setup, which i believe is the case, then we should put this in requirements.txt file.
For context, CLI depends on msrestazure to pull in the adal which helps avoid errors caused by diamond dependency
//CC: @lmazuel

Comment (yugangw-msft) [reply src/azure-cli-core/setup.py]: On a second thought, CLI needs adal with 1.0.2+ , so i don't mind calling out here, but let us listen to Laurent

Comment (lmazuel) [reply src/azure-cli-core/setup.py]: It's fine. Out of curiosity, why do you need 1.0.2? What is interesting inside?

Comment (lmazuel) [reply src/azure-cli-core/setup.py]: Ok, I read the ChangeLog of ADAL. LGTM.

**Pull request merged**